### PR TITLE
refactor angular config for better dev mode

### PR DESCRIPTION
**Pull request by dwradcliffe**
Before this, doing rapid development on the frontend required rebuilding the whole thing which would take several minutes each time. With this change, we can run the frontend server, watch for changes, and get a compiled result in ~1 second.

Note: I'm still not able to get HMR working but this is still loads better.

This also cleans up the index.html between the 3 flavors so that we don't have errors in docker due to that desktop library.

There are still a few weird errors in the console when developing but those can be another follow up.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -86,23 +86,19 @@ Next we'll start the processes described above:
...
-# eg. make build-frontend-sandbox
+make dev-frontend
 
 # In terminal #2, run the following
 make serve-backend
 
 ```
 
 _Note_: Fasten can run in 2 modes: sandbox or production (prod). In sandbox mode, it can only communicate with test servers (full of synthetic health data).
+By default, Fasten the dev environment will run in sandbox mode.
```
Comment (hkamran80) [inline CONTRIBUTING.md:97]: Does "Fasten" need to be in here?

Comment (dwradcliffe) [reply CONTRIBUTING.md]: good catch, thanks!

Comment (hkamran80) [reply CONTRIBUTING.md]: Happy to help!

**Comment by AnalogJ** [comment]
Hey @dwradcliffe Apologies, I should have taken a look at this PR earlier.

hm. the intention of the `serve-frontend-*` Make commands is to do the same thing as the `dev-frontend` command you just added. Could we just merge your changes there?

**Comment by AnalogJ** [comment]
Also that `proxy` stuff is great, I didn't realize Angular had that kind of functionality. Thanks for bringing it to my attention!

**Review by AnalogJ — APPROVED** [review]

**Pull request merged**